### ISSUE TITLE
Problem with validation of keys in dict type 

### ISSUE THREAD

**Issue by tadeu** [nicolaiarocci/cerberus#237]
I'm trying to validate a dict type where the keys may be any string. This is a simple example:

```python
schema = {
    'test': {
        'type': 'dict',
        'keyschema': {'type': 'string'},
        'valueschema': {'type': 'integer'},
    },
}

v = Validator(schema)
v.validate({'test': {'a': 1, 'b': 2}})
print v.errors
```

Output:
```
WARNING cerberus: `keyschema` is deprecated, use `valueschema` instead
{'test': [{'a': 'must be of string type'}, {'b': 'must be of string type'}]}
```

(Using Cerberus 0.9.2)
What is wrong? Thanks!

**Comment by funkyfuture** [comment]
in this version `keyschema` is an alias of `valueschema` for historical reasons.

either use `propertyschema` instead or use the current `master`-branch's `HEAD`.

and always rtfm, especially when confronted with an obvious warning.

**Comment by tadeu** [comment]
Ok, I've "RTFM", but it is confusing. So let me rephrase the problem.
How can I validate documents like this:

```
{'test': {'a': 1, 'b': 2, 'c': 3}}
```

Where `test` is required, and it is a dict composed of any number of key/values, but the keys must follow a schema (in this case, to be a simple string), and the values must follow a different schema (in this case, to be an integer).

**Comment by funkyfuture** [comment]
did you read the `stable` or the `latest` docs? which version do you intend to use? what is confusing in particular?

**Comment by tadeu** [comment]
In both, but I was trying to accomplish this with the `stable` version.

For example, this is confusing in both documentations:

> ### `keyschema`
> This is the counterpart to `valueschema` that validates the _keys_ of a dict. **For historical reasons it is not named `keyschema`.**

Also it is not clear in the stable documentation how to accomplish what I want, but at least it has got this not in the `latest` documentation:

> ### Note
> To validate arbitrary keys of a mapping, see `keyschema`, resp. `valueschema` for validating arbitrary values of a mapping.

Well... guess I'll have to try working on `master`...

**Comment by funkyfuture** [comment]
lol, that was rather a search & destroy and not replace when i recently updated the docs. thanks for pointing that out.

your example should work as expected with the current master.

Am 11. Mai 2016 17:41:52 MESZ, schrieb Tadeu Manoel <notifications@github.com>:
